Automattic/jetpack
# Try/google maps api key

API for setting/updating/deleting a persisted Google Maps API key which is used by the Map block. This PR is for testing only.

~~JN: https://jurassic.ninja/create/?gutenberg&gutenpack&shortlived&jetpack-beta&calypsobranch=atavist/map-block-api-key-integration&branch=try/google-maps-api-key~~

## Testing Instructions

_IMPORTANT NOTE: These testing notes apply to the editor only. If you create a map block following these instructions and then publish or preview, nothing will show up. This is because of an unrelated change needed in Jetpack, which will be included in a new PR after this one is merged._

1) Spin up a JN instance with this link: https://jurassic.ninja/create/?gutenberg&gutenpack&shortlived&jetpack-beta&calypsobranch=master&branch=try/google-maps-api-key
2) Click `Set up Jetpack`
3) Click `Start with free`
4) Click `Activate recommeded features`
5) Navigate to `Settings->Jetpack Constants`
6) Check the box by `JETPACK_BETA_BLOCKS` and `Save Changes`
7) Create new Post
8) Click to add a block, search for `map`, add
9) Paste in a bogus Mapbox key (e.g. `notarealmapboxkey`). You should get an `Invalid API key` error message. This tests the validation code in the PUT method.
10) Paste in a real Mapbox key. You can use this key, which I'll deactivate as soon as the branch is merged: `pk.eyJ1IjoicmFiYmVyc29uIiwiYSI6ImNqb2htaWh1dTAxNnYza250dGc5N2FiamUifQ.hEAdji_oZcG0NI9VBswspw` - this should be accepted and the Map should load. This demonstrates the PUT method.
11) Click on the block, make sure Block Settings sidebar is visible, and open the `Mapbox Access Token` Panel.
12) Click `Remove Key`. This should return the block to initial state. This is the DELETE method in action.
13) Paste the real key back in.
14) Open `Mapbox Access Token` panel again, alter the token and click `Update Key`. This should show the `Invalid API Key` error and return the block to its initial state.
